### PR TITLE
Add --prepare option to migrate command

### PR DESCRIPTION
**Pull request by nadvornik**
## What does this PR change?

This PR adds `--prepare` option to `migrate` command.
With this option, the migration does not stop the services on source server and ends after syncing
the data.

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added

- [ ] **DONE**

## Links

Issue(s): https://github.com/uyuni-project/uyuni-tools/issues/181

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: mgradm/cmd/migrate/kubernetes/utils.go [hunk 1]
```diff
@@ -90,6 +90,11 @@ func migrateToKubernetes(
 		return utils.Errorf(err, L("cannot run migration"))
 	}
 
+	if flags.Prepare {
+		log.Info().Msg(L("Migration prepared. Run the 'migrate' command without '--prepare' to finish the migration."))
```
Comment (cbosdo) [inline mgradm/cmd/migrate/kubernetes/utils.go:94]: Should we scale to 0 after this? Is the migration without `--prepare` coping well with the existing resources?

File: mgradm/cmd/migrate/podman/utils.go [hunk 2]
```diff
@@ -35,10 +35,14 @@ func migrateToPodman(globalFlags *types.GlobalFlags, flags *podmanMigrateFlags,
 	sshAuthSocket := migration_shared.GetSshAuthSocket()
 	sshConfigPath, sshKnownhostsPath := migration_shared.GetSshPaths()
 
-	tz, oldPgVersion, newPgVersion, err := podman.RunMigration(serverImage, flags.Image.PullPolicy, sshAuthSocket, sshConfigPath, sshKnownhostsPath, sourceFqdn, flags.User)
+	tz, oldPgVersion, newPgVersion, err := podman.RunMigration(serverImage, flags.Image.PullPolicy, sshAuthSocket, sshConfigPath, sshKnownhostsPath, sourceFqdn, flags.User, flags.Prepare)
 	if err != nil {
 		return utils.Errorf(err, L("cannot run migration script"))
 	}
+	if flags.Prepare {
+		log.Info().Msg(L("Migration prepared. Run the 'migrate' command without '--prepare' to finish the migration."))
```
Comment (cbosdo) [inline mgradm/cmd/migrate/podman/utils.go:43]: Same here: should we stop the service here? Is `migrate` behaving correctly with the existing resources?

**Review by cbosdo — COMMENTED** [review]
LGTM, with some questions about whether to stop the services?

File: mgradm/cmd/migrate/podman/utils.go (hunk 2 above)
Comment (nadvornik) [reply mgradm/cmd/migrate/podman/utils.go:43]: There is nothing to stop for podman. The container exists only while the migration script is running.

File: mgradm/cmd/migrate/kubernetes/utils.go (hunk 1 above)
Comment (nadvornik) [reply mgradm/cmd/migrate/kubernetes/utils.go:94]: I have to test the kubernetes part.

**Comment by rjmateus** [comment]
I if we set the prepare command we should stop just after the rsync of each volume. All the config changes (and probably the distribution copy) should be done only on final migrate. 
The goal is to start copy the major data folder to speed migration.

**Comment by nadvornik** [comment]
I think it is safe to let the migration scrip run till the end. All the config changes it does will be overwritten by the final run, but it can help to spot potential problems earlier, before stopping the old server.

**Comment by nadvornik** [comment]
I have added kubernetes fixes -  scale to 0 and skip traefik installation.

**Review by cbosdo — APPROVED** [review]

**Pull request closed without merging**